src/OrleansBondUtils/BondSerializer.cs

```diff
@@ -191,15 +197,35 @@ private void Register(Type type)
             var clonerType = typeof(Cloner<>);
             var realType = clonerType.MakeGenericType(type);
             var clonerInstance = Activator.CreateInstance(realType);
-            var cloneMethod = realType.GetMethod("Clone").MakeGenericMethod(type);
-            var copierDelegate = cloneMethod.CreateDelegate(
-                    typeof(Func<,>).MakeGenericType(new[] { type, type }),
-                    clonerInstance);
             var serializer = new BondTypeSerializer(type);
             var deserializer = new BondTypeDeserializer(type);
-            CopierDictionary.TryAdd(type.TypeHandle, copierDelegate);
+            var parameterExpression1 = Expression.Parameter(typeof(object));
```

Comment:
I might be asking a stupid question, but as far as I understand, your previous code to create the cloner - it was using reflection only on the first time (first creation). At invocation time it was just invoking the delegate.
So is the new code more efficient at the invocation time? If it is the same and the only difference is the creation time, I would argue the previous code was better, since humans like me could actually understand it.
I even opted for use dynamic keyword in [Protobuf serializer](https://github.com/dotnet/orleans/blob/master/src/OrleansGoogleUtils/ProtobufSerializer.cs#L58): you really cannot beat the one line simplicity of it.
And as far as I understood, at invoke time there won't be any differences in perf, since in both cases the reflection info is cached. You probably can't use dynamic in your case, since I had an actual instance while you have to create one, but my question is about old code vs. new code.

But it is possible I misunderstood how it all works. I am not an expert of reflection and know very little about expression trees.